src/js/structs/Job.js

```diff
@@ -0,0 +1,21 @@
+import Item from './Item';
+
+module.exports = class Job extends Item {
+  constructor() {
+    super(...arguments);
+
+    const id = this.getId();
+    if (id !== '/' && (!id.startsWith('/') || id.endsWith('/'))) {
+      throw new Error(`Id (${id}) must start with a leading slash ("/") and should not end with a slash, except for root id which is only a slash`);
```

Comment:
Your comment actually spawned a larger conversation about what we should and should not do in the structs. We decided to move the whole building of the tree to a 'reducer'-like function that is being used in the actions before the struct is being created. That way we can handle any malformed data gracefully before handing it off to the struct:
https://github.com/dcos/dcos-ui/pull/246/files#diff-976ccf0fd4e369643bb74528d0c47785R20